File: app/src/main/java/ru/andrey/cleandictionary/di/translation/TranslationComponent.java
package ru.andrey.cleandictionary.di.translation;

import dagger.Component;
import ru.andrey.cleandictionary.di.app.AppComponent;
import ru.andrey.cleandictionary.presentation.presenter.AddWordPresenter;
import ru.andrey.cleandictionary.presentation.presenter.DictionaryListPresenter;

@Component(dependencies = AppComponent.class, modules = {TranslationModule.class})
@TranslationScope
public interface TranslationComponent {

	AddWordPresenter getAddWordPresenter();

	DictionaryListPresenter getDictionaryListPresenter();
}

File: presentation/src/main/java/ru/andrey/cleandictionary/di/translation/TranslationComponent.kt
package ru.andrey.cleandictionary.di.translation

import dagger.Component
import dagger.Subcomponent
import ru.andrey.cleandictionary.di.ApplicationComponent
import ru.andrey.cleandictionary.di.scope.Feature
import ru.andrey.cleandictionary.di.scope.Screen
import ru.andrey.cleandictionary.presentation.presenter.addword.AddWordPresenter
import ru.andrey.cleandictionary.presentation.presenter.list.WordListPresenter
import ru.andrey.domain.repository.TranslationRepository

@Feature
@Component(dependencies = [ApplicationComponent::class],
        modules = [TranslationDataModule::class, TranslationModule::class])
interface TranslationComponent {

    fun translationRepository(): TranslationRepository

    fun presenter(): PresenterComponent

    @Screen
    @Subcomponent
    interface PresenterComponent {

        fun wordListPresenter(): WordListPresenter

        fun addWordPresenter(): AddWordPresenter
    }
}
